src/graph/entities/qg_node.c

```diff
@@ -46,6 +55,16 @@ int QGNode_EdgeCount(const QGNode *n) {
 void QGNode_ConnectNode(QGNode *src, QGNode *dest, QGEdge *e) {
 	src->outgoing_edges = array_append(src->outgoing_edges, e);
 	dest->incoming_edges = array_append(dest->incoming_edges, e);
+
+	// Set src node as highly connected if in-degree + out-degree > 2
+	if(src->highly_connected == false && QGNode_Degree(src) > 2) {
+		src->highly_connected = true;
+	}
```

Comment:
Instead of this logic, it might be preferable to replace `highly_connected` with an `int node_degree` or similar to reduce the number of conditionals. Not a significant optimization, though.

Reply:
I think that once a node is cloned, we discard all its connections, (incoming and outgoing) but maintain the `highly_connected` attribute, I think it would be a bit more confusing to see a node without any edges but with a degree of say 3, not that seeing `highly_connected` set to true, for that same ndoe.

Reply:
:+1: Good point!